File: backend/backend-prod/src/main/java/com/proof/concept/model/UserPasswordChangeModel.java
package com.proof.concept.model;

import java.io.Serializable;

public class UserPasswordChangeModel implements Serializable {

	private static final long serialVersionUID = 1L;
	private int userId;
	private String oldPassword;
	private String newPassword;

	public UserPasswordChangeModel() {
	}

	public int getUserId() {
		return userId;
	}

	public void setUserId(int userId) {
		this.userId = userId;
	}

	public String getOldPassword() {
		return oldPassword;
	}

	public void setOldPassword(String oldPassword) {
		this.oldPassword = oldPassword;
	}

	public String getNewPassword() {
		return newPassword;
	}

	public void setNewPassword(String newPassword) {
		this.newPassword = newPassword;
	}

}

File: frontend/src/app/additional-models/UserPasswordChangeModel.ts
export interface IUserPasswordChangeModel {
    userId?: number;
    oldPassword?: string;
    newPassword?: string;
  }
  
  export  class  UserPasswordChangeModel implements IUserPasswordChangeModel {
    constructor(
      public userId?: number,
      public oldPassword?: string,
      public newPassword?: string
    ) {}
  }